hack/test_e2e.go

```diff
@@ -0,0 +1,944 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"log"
+	"bytes"
+	"strconv"
+	"strings"
+	"path"
+	"text/template"
+) 
+
+const (
+	KIND_VERSION="v0.17.0"
+	K8S_VERSION="v1.25.3"
+	KPNG_IMAGE_TAG_NAME="kpng:test_270323_0904" 
+	CLUSTER_CIDR_V4="10.1.0.0/16"
+	SERVICE_CLUSTER_IP_RANGE_V4="10.2.0.0/16"
+	KINDEST_NODE_IMAGE="docker.io/kindest/node"
+	E2E_K8S_VERSION="v1.25.3"
+	KUBECONFIG_TESTS="kubeconfig_tests.conf"
+)
+
+// System data
+const (
+	NAMESPACE="kube-system"
+	SERVICE_ACCOUNT_NAME="kpng"
+	CLUSTER_ROLE_BINDING_NAME="kpng"
+	CLUSTER_ROLE_NAME="system:node-proxier"
+	CONFIG_MAP_NAME="kpng"
+
+	KPNG_SERVER_ADDRESS="unix:///k8s/proxy.sock"
+	KPNG_DEBUG_LEVEL="4"
+)
+
+var kpng_dir, CONTAINER_ENGINE string
+
+func if_error_exit(error_msg error) {
+    // Description:
+    // Validate if previous command failed and show an error msg (if provided) 
+    //
+	// Arguments:
+	// $1 - error message if not provided, it will just exit
+	///////////////////////////////////////////////////////////////////////////
+	if error_msg != nil {
+		log.Fatal(error_msg)
+	}
+} 
+
+func command_exist(cmd_test string) bool {
+	///////////////////////////////////////////////////////////////////////////
+	// Description:                                                            
+    // Checkt if a binary exists                                               
+    //                                                                         
+    // Arguments:                                                              
+    //   arg1: binary name                                                     
+	///////////////////////////////////////////////////////////////////////////
+	
+	cmd_script := "command -v " + cmd_test + " > /dev/null 2>&1"
+	cmd := exec.Command("bash", "-c", cmd_script)
+	err := cmd.Run()
+	
+	if err == nil {
+		return true
+	}
+	return false
+}
+
+func add_to_path(directory string) {
+    // Description:                                                            //
+    // Add directory to path                                                   //
+    //                                                                         //
+    // Arguments:                                                              //
+    //   arg1:  directory                                                      //	
+	///////////////////////////////////////////////////////////////////////
+	_, err := os.Stat(directory)
+	if err != nil && os.IsNotExist(err) {
+		if_error_exit(err)
+	}
+
```

Comment:
Hi @aroradaman, after reading the link - [](https://pkg.go.dev/k8s.io/klog/v2) - I still don't understand why we should use logger instead of Println. At first I thought you were talking about the **Log** package! But then I realized that it's an all different package! Please can you help me understanding why **klog** is a better option? And wouldn't **Log** work here?

Reply:
@emabota klog will give you the option for structured logging and you can control the verbosity ... also the logging will be consistent through kpng